perllib/FixMyStreet/Roles/DDProcessor.pm

```diff
@@ -316,14 +321,37 @@ sub _duplicate_waste_report {
 sub _process_reference {
     my ($self, $payer) = @_;
 
-    (my $uprn = $payer) =~ s/^GGW//;
+    # Old style GGW references
+    if ((my $uprn = $payer) =~ s/^GGW//) {
+        my $len = length($uprn);
+        my $rs = FixMyStreet::DB->resultset('Problem')->search({
+            extra => { like => '%uprn,T5:value,I' . $len . ':'. $uprn . '%' },
+        }, {
+            order_by => [ { -desc => 'created' }, { -desc => 'id' } ],
+        })->to_body( $self->body );
```

Comment:
If I'm reading this right it will find the most recent report to Bromley/whoever that has the right `uprn` value in its extra regardless of category etc. I think it might be worth tightening up the search criteria if at all possible just to reduce the chance of some future bug if e.g. an unrelated category added a `uprn` extra or something.

Reply:
Looks like it always does a further search for category after it's returned. Not sure why it can't do it here, though, you're right

Reply:
Never sure about leaving private issue links in public commits, but yeah, should probably do so here, and give better commit messages (looks like I managed 1 out of 4?).

Reply:
Yeah, it's not super helpful for anyone without access to the issue in question but it doesn't give anything private away (IMO) and most devs who work on the codebase do have access.

Reply:
Just to note for future, one issue with it is if you do lots of rebasing of a commit with an issue ref contained, each rebased commit appears on that issue's page as a commit pointing at it, for ever - see e.g. https://github.com/mysociety/societyworks/issues/676 for an extreme case. Would be good if you could associate a PR description with the commits somehow!